drivers/espi/espi_npcx.c

```diff
@@ -865,6 +865,14 @@ static int espi_npcx_init(const struct device *dev)
 		return ret;
 	}
 
+	if (IS_ENABLED(CONFIG_ESPI_NPCX_BYPASS_CH_ENABLE_FATAL_ERROR)) {
```

Comment:
Was wondering, do we need a dedicated Kconfig option for this? Maybe it could be turned on based on the device family or runtime detection of affected devices? I see Nordic has ha file in their hal and they turn those on and off based on other options (https://github.com/zephyrproject-rtos/hal_nordic/blob/master/nrfx/mdk/nrf52_erratas.h) thinking that something like that may make it less likely that we unintentionally leave the wokaround off.

Reply:
Thank you for the comment.
Ths symbol ESPI_NPCX_BYPASS_CH_ENABLE_FATAL_ERROR a a [invisible symbol](https://docs.zephyrproject.org/2.6.0/guides/kconfig/setting.html#visible-and-invisible-kconfig-symbols) (i.e. a symbol without the prompt).
It it not shown in the interactive configuration interfaces, and users have no direct control over their value.
The user should not  have a chance to turn it off if he doesn't modify the depends or default in the Kconfig.npcx directly.
I followed the precedence for the similiar [approach](https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/silabs_exx32/Kconfig#L200) to use Kconfig for the errata.

Reply:
Right but in the efr32 case they keep the option off by default and turn it on selectively for few chips on the series in a family driver, while in thise case we have the option depending on the whole family (`depends on SOC_FAMILY_NPCX`) and the driver is family specific anyway (it also depends on `SOC_FAMILY_NPCX`)... so in the current scenario there's no point in having the option at all. Do you forsee this being applied only for few chips?

Reply:
I expect our next generation chip should fix the issue. When we introduce the new chip and confirm this bug is fixed in the future, I will revisit this symbol and modify it to  depend on NPCX7&NPCX9 only.

Reply:
Cool with that... how about making it depend on NPCX7 and NPCX9 now?

Reply:
Done.